### PR TITLE
[#48275] Default work packages PDF export styling

### PR DESCRIPTION
**Pull request by as-op**
https://community.openproject.org/work_packages/48275

### CODE REVIEW

**Comment by as-op** [comment]
Hi @klaustopher, may I kindly ask you for a follow-up review? This PR extracts the hardcoded styles into a yml-file and adds schema validation for it.
The default styling will probably stay in the yml, while user created styling will be stored in and retrieved from the db.

File: app/models/work_package/pdf_export/common.rb [hunk 1]
```diff
@@ -111,15 +111,15 @@ def get_subject_column_cell(work_package, value)
...
+    "<color rgb='#{styles.link_color}'><link href='#{href}'>#{caption}</link></color>"
   end
 
   def make_link_anchor(anchor, caption)
     "<link anchor=\"#{anchor}\">#{caption}</link>"
   end
 
-  def align_to_left_position(text, align, style)
-    text_width = pdf.width_of(text, style)
+  def align_to_left_position(text, align, text_style)
```
Comment (klaustopher) [inline app/models/work_package/pdf_export/common.rb:121]: 👍 for better naming of variables

File: app/models/work_package/pdf_export/overview_table.rb [hunk 2]
```diff
@@ -143,17 +143,17 @@ def build_nr_cell(work_package, id_wp_meta_map)
   end
 
   def build_subject_cell(content, work_package, id_wp_meta_map)
-    opts = overview_table_cell_padding_style
-    padding_left = opts[:padding_left]
+    opts = styles.overview_table_cell
+    padding_left = opts[:padding_left] || 0
```
Comment (klaustopher) [inline app/models/work_package/pdf_export/overview_table.rb]: :memo: When you want to provide a default value for a hash lookup you can also use 

```suggestion
    padding_left = opts.fetch(:padding_left, 0)
```

To some that looks nicer. No need to change, just pointing it out

File: app/models/work_package/pdf_export/standard.yml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      no_border_top: true
+    image:
+      align: "left"
+    codeblock:
+      background_color: "F5F5F5"
+      color: "880000"
+      padding: "3mm"
+      size: 8
+      margin_top: "2mm"
+      margin_bottom: "2mm"
```
Comment (klaustopher) [inline app/models/work_package/pdf_export/standard.yml]: ❓ Is there a reason we are adding units here and no tfor the other margins?

**Review by klaustopher — APPROVED** [review]
LGTM!

File: app/models/work_package/pdf_export/standard.yml (hunk 3 above)
Comment (as-op) [reply app/models/work_package/pdf_export/standard.yml]: No good reason other than it's possible to use units. ^^
I unified the usage to have non specified. (PostscriptPoint is the default)

**Pull request merged**